hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -91,412 +89,722 @@ public class SingleTransactionRecordBuilderImpl
-                                serialNumbers),
-                        transactionHash,
-                        consensusTimestamp,
-                        transactionID,
-                        memo,
-                        transactionFee,
-                        body,
-                        transferList,
-                        tokenTransferLists,
-                        scheduleRef,
-                        assessedCustomFees,
-                        automaticTokenAssociations,
-                        parentConsensusTimestamp != null ? HapiUtils.asTimestamp(parentConsensusTimestamp) : null,
-                        alias,
-                        ethereumHash,
-                        paidStakingRewards,
-                        entropy,
-                        evmAddress),
-                transactionSidecarRecords);
+
+        return new SingleTransactionRecord(transaction, transactionRecord, transactionSidecarRecords);
     }
 
     // ------------------------------------------------------------------------------------------------------------------------
     // base transaction data
-    public SingleTransactionRecordBuilderImpl transaction(Transaction transaction) {
-        this.transaction = transaction;
+
+    /**
+     * Sets the transaction.
+     *
+     * @param transaction the transaction
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl transaction(@NonNull final Transaction transaction) {
+        this.transaction = requireNonNull(transaction, "transaction must not be null");
         return this;
     }
 
-    public SingleTransactionRecordBuilderImpl transactionBytes(Bytes transactionBytes) {
-        this.transactionBytes = transactionBytes;
+    /**
+     * Sets the transaction bytes that will be used to compute the transaction hash.
+     *
+     * @param transactionBytes the transaction bytes
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl transactionBytes(@NonNull final Bytes transactionBytes) {
+        this.transactionBytes = requireNonNull(transactionBytes, "transactionBytes must not be null");
         return this;
     }
 
-    public SingleTransactionRecordBuilderImpl transactionID(TransactionID transactionID) {
-        this.transactionID = transactionID;
+    /**
+     * Sets the transaction ID.
+     *
+     * @param transactionID the transaction ID
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl transactionID(@NonNull final TransactionID transactionID) {
+        requireNonNull(transactionID, "transactionID must not be null");
+        this.transactionRecordBuilder.transactionID(transactionID);
         return this;
     }
 
-    public SingleTransactionRecordBuilderImpl memo(String memo) {
-        this.memo = memo;
+    /**
+     * Sets the memo.
+     *
+     * @param memo the memo
+     * @return the builder
+     */
```

Comment:
More general comment since I see we have this in other places as well - do we need to add that kind of comments? The comment seems to not provide more information than it's already known when we see the method's signature and I think it's a bit redundant.

Reply:
You are right, such comments do not add much value. But once we enforce that all public methods have javadocs (which we should), it would become cumbersome and error-prone to configure the validator with such exceptions. It is much easier to add some short javadocs.